cfme/tests/middleware/random_methods.py

```diff
@@ -7,50 +7,88 @@
-    assert len(server_groups) > 0, "There is no server_groups(s) available in UI"
+def get_random_server_group(provider, load_from="ui"):
+    load_from = load_from.lower()
+    if load_from == "ui":
+        server_groups = MiddlewareServerGroup.server_groups(get_random_domain(provider, load_from))
+    elif load_from == "db":
+        server_groups = MiddlewareServerGroup.server_groups_in_db(
+            get_random_domain(provider, load_from))
+    elif load_from == "mgmt":
+        server_groups = MiddlewareServerGroup.server_groups_in_mgmt(
+            get_random_domain(provider, load_from))
+    assert len(server_groups) > 0, "There is no server group(s) available in {}".format(load_from)
     return get_random_list(server_groups, 1)[0]
 
 
-def get_random_deployment(provider):
-    deployments = MiddlewareDeployment.deployments(provider=provider)
-    assert len(deployments) > 0, "There is no deployment(s) available in UI"
+def get_random_deployment(provider, load_from="ui"):
+    load_from = load_from.lower()
+    if load_from == "ui":
+        deployments = MiddlewareDeployment.deployments(provider=provider)
+    elif load_from == "db":
+        deployments = MiddlewareDeployment.deployments_in_db(provider=provider)
+    elif load_from == "mgmt":
+        deployments = MiddlewareDeployment.deployments_in_mgmt(provider=provider)
+    assert len(deployments) > 0, "There is no deployment(s) available in {}".format(load_from)
     return get_random_list(deployments, 1)[0]
 
 
-def get_random_datasource(provider):
-    datasources = MiddlewareDatasource.datasources(provider=provider)
-    assert len(datasources) > 0, "There is no datasource(s) available in UI"
+def get_random_datasource(provider, load_from="ui"):
+    load_from = load_from.lower()
+    if load_from == "ui":
+        datasources = MiddlewareDatasource.datasources(provider=provider)
+    elif load_from == "db":
+        datasources = MiddlewareDatasource.datasources_in_db(provider=provider)
+    elif load_from == "mgmt":
+        datasources = MiddlewareDatasource.datasources_in_mgmt(provider=provider)
+    assert len(datasources) > 0, "There is no datasource(s) available in {}".format(load_from)
     return get_random_list(datasources, 1)[0]
 
 
-def get_random_messaging(provider):
-    messagings = MiddlewareMessaging.messagings(provider=provider)
-    assert len(messagings) > 0, "There is no messaging(s) available in UI"
+def get_random_messaging(provider, load_from="ui"):
+    load_from = load_from.lower()
+    if load_from == "ui":
+        messagings = MiddlewareMessaging.messagings(provider=provider)
+    elif load_from == "db":
+        messagings = MiddlewareMessaging.messagings_in_db(provider=provider)
+    elif load_from == "mgmt":
+        messagings = MiddlewareMessaging.messagings_in_mgmt(provider=provider)
+    assert len(messagings) > 0, "There is no messaging(s) available in {}".format(load_from)
```

Comment:
If load_from doesn't match any of the 3 conditions, this will throw an exception as messagings is not defined.
Consider adding else clause to cleanly catch this error.